internal/pkg/cli/pipeline_deploy.go

```diff
@@ -258,13 +270,71 @@ func (o *deployPipelineOpts) Execute() error {
 		PermissionsBoundary: o.app.PermissionsBoundary,
 	}
 
+	if o.showDiff {
+		tpl, err := o.pipelineStackConfig(deployPipelineInput).Template()
+		if err != nil {
+			return fmt.Errorf("stack template local: %w", err)
+		}
+		if err = o.diff(deployPipelineInput, tpl); err != nil {
+			var errHasDiff *errHasDiff
+			if !errors.As(err, &errHasDiff) {
+				return err
+			}
+		}
+		if !o.skipConfirmation {
+			contd, err := o.prompt.Confirm(continueDeploymentPrompt, "")
+			if err != nil {
+				return fmt.Errorf("ask whether to continue with the deployment: %w", err)
+			}
+			if !contd {
+				return nil
+			}
+		}
+	}
 	if err := o.deployPipeline(deployPipelineInput); err != nil {
 		return err
 	}
-
 	return nil
 }
 
+// PipelineDeployDiff returns the stringified diff of the template against the deployed template of the pipeline.
+func (o *deployPipelineOpts) pipelineDeployDiff(in *deploy.CreatePipelineInput, template string) (string, error) {
+	tmpl, err := o.pipelineDeployer.Template(stack.NameForPipeline(in.AppName, in.Name, in.IsLegacy))
+	if err != nil {
+		var errNotFound *awscloudformation.ErrStackNotFound
+		if !errors.As(err, &errNotFound) {
+			return "", fmt.Errorf("retrieve the deployed template for %q: %w", in.Name, err)
+		}
+		tmpl = ""
+	}
+	diffTree, err := templatediff.From(tmpl).ParseWithCFNOverriders([]byte(template))
+	if err != nil {
+		return "", fmt.Errorf("parse the diff against the deployed pipeline stack %q: %w", in.Name, err)
+	}
+	buf := strings.Builder{}
+	if err := diffTree.Write(&buf); err != nil {
+		return "", err
+	}
+	return buf.String(), nil
+}
+
+func (o *deployPipelineOpts) diff(in *deploy.CreatePipelineInput, tpl string) error {
+
+	if out, err := o.pipelineDeployDiff(in, tpl); err != nil {
+		var errHasDiff *errHasDiff
+		if !errors.As(err, &errHasDiff) {
```

Comment:
I wonder if we should do a switch here instead of the `else if`s.

Reply:
We can do that too but since there are no large number of comparisons and I am not just doing the simple value comparisons instead performing operations which are involving function calls and error handling. Wouldn't it be better if we use if/else if in this case than using switch 🤔?

Reply:
Can we reuse https://github.com/aws/copilot-cli/blob/b408b6d0a833d5e9be7f560e2296e24014e63fb5/internal/pkg/cli/svc_deploy.go#L604